src/main/java/org/openstreetmap/atlas/generator/AtlasGenerator.java

```diff
@@ -354,4 +299,17 @@ protected SwitchList switches()
         result.addAll(AtlasGeneratorParameters.switches());
         return result;
     }
+
+    @SuppressWarnings("unchecked")
+    private void saveAsHadoop(final JavaPairRDD<?, ?> atlasRDD, final AtlasGeneratorJobGroup group,
+            final String output)
+    {
+        this.getContext().setJobGroup(group.getId().toString(), group.getDescription());
+        atlasRDD.saveAsHadoopFile(getAlternateSubFolderOutput(output, group.getCacheFolder()),
+                Text.class, group.getKeyClass(),
+                (Class<? extends MultipleOutputFormat<?, ?>>) group.getOutputClass(),
```

Comment:
So many questions... 🤔

Reply:
this block of code was repeated throughout `start()`, but with different constants being applied. the constants all varied based on the task being done, so while this is a much more generic block of code, I think it's cleaner -- now, to figure out what each task is doing, we're managing those constants explicitly instead of just hard coding them into the flow

Reply:
unrelated but also this is slightly out of date, I will update to fix the unchecked and casting here